lib/puppet/application.rb

```diff
@@ -281,6 +281,15 @@ def run_mode( mode_name = nil)
       @run_mode = Puppet::Util::RunMode[ mode_name || Puppet.settings.preferred_run_mode ]
     end
 
+    def environment_mode(mode_name)
+      raise Puppet::Error, _("Invalid environment mode '%{mode_name}'") % { mode_name: mode_name } unless [:local, :remote, :not_required].include?(mode_name)
+      @environment_mode = mode_name.to_s.downcase.intern
```

Comment:
Why use `to_s.downcase.intern`? We know it's a symbol already from the line above.

Reply:
Oh, that's a good point. I both copied the check from the run_mode verification and kept it in just in case the user decided to not use a symbol. Do you think it would be better to take it out, or add the conversion in above the raise?

Reply:
I'd rather do less magic conversion. If we expect a symbol, let's make sure people pass in a symbol. I would also add a docs comment to this method stating that we expect to be given a symbol.

Reply:
:+1: